### PR TITLE
[sumologicexporter] Enable exporter

### PR DESCRIPTION
**Pull request by sumo-drosiek**
**Description:**

- Enable sumologic exporter
- Minor improvement for compress
- Use github.com/pkg/errors for errors
- Add context propagation to sender
- Add support for dots in headers extrapolation

**Link to tracking Issue:**
#1498 

**Testing:**
- Unit tests
- Manual e2e tests

**Documentation:**
- Comments

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
Already after internal review
cc: @pmm-sumo @pmalek-sumo

**Review by pmalek-sumo — APPROVED** [review]
👍🏻

File: exporter/sumologicexporter/exporter.go [hunk 1]
```diff
@@ -16,41 +16,45 @@ package sumologicexporter
 
 import (
 	"context"
-	"errors"
-	"fmt"
+	"net/http"
 
+	"github.com/pkg/errors"
```
Comment (bogdandrutu) [inline exporter/sumologicexporter/exporter.go:21]: Please avoid using this pkg :)

Comment (pmalek-sumo) [reply exporter/sumologicexporter/exporter.go:21]: Can you explain why?

Quick search reveals numerous usages of this package: https://github.com/open-telemetry/opentelemetry-collector-contrib/search?q=github.com%2Fpkg%2Ferrors in other exporters and receivers.

Any other alternative is suggested to be used in otelcol repositories?

Comment (bogdandrutu) [reply exporter/sumologicexporter/exporter.go:21]: As you can see even in this file `fmt.Errorf` has exactly the same API and is a language native method.

Comment (sumo-drosiek) [reply exporter/sumologicexporter/exporter.go]: Done

**Comment by jpkrohling** [comment]
I'm removing myself as assignee for this issue, as I'm unavailable until Jan 11.

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Please rebase

**Comment by sumo-drosiek** [comment]
@bogdandrutu rebased

**Comment by tigrannajaryan** [comment]
All comments are resolved and the PR is approved. Merging.

**Pull request merged**